// @affinda/affinda/dist-esm/models/index.js
/** Known values of {@link ResumeSkillSourcesItemSection} that the service accepts. */
export var KnownResumeSkillSourcesItemSection;
(function (KnownResumeSkillSourcesItemSection) {
    KnownResumeSkillSourcesItemSection["Achievements"] = "Achievements";
    KnownResumeSkillSourcesItemSection["AdditionalInformation"] = "AdditionalInformation";
    KnownResumeSkillSourcesItemSection["Education"] = "Education";
    KnownResumeSkillSourcesItemSection["Extracurriculars"] = "Extracurriculars";
    KnownResumeSkillSourcesItemSection["Organisations"] = "Organisations";
    KnownResumeSkillSourcesItemSection["Other"] = "Other";
    KnownResumeSkillSourcesItemSection["PersonalDetails"] = "PersonalDetails";
    KnownResumeSkillSourcesItemSection["Projects"] = "Projects";
    KnownResumeSkillSourcesItemSection["Publications"] = "Publications";
    KnownResumeSkillSourcesItemSection["Referees"] = "Referees";
    KnownResumeSkillSourcesItemSection["Skills"] = "Skills";
    KnownResumeSkillSourcesItemSection["Summary"] = "Summary";
    KnownResumeSkillSourcesItemSection["Training"] = "Training";
    KnownResumeSkillSourcesItemSection["WorkExperience"] = "WorkExperience";
    KnownResumeSkillSourcesItemSection["NotPopulated"] = "NotPopulated";
    KnownResumeSkillSourcesItemSection["Header"] = "Header";
    KnownResumeSkillSourcesItemSection["Footer"] = "Footer";
})(KnownResumeSkillSourcesItemSection || (KnownResumeSkillSourcesItemSection = {}));
/** Known values of {@link Enum1} that the service accepts. */
export var KnownEnum1;
(function (KnownEnum1) {
    KnownEnum1["Resumes"] = "resumes";
    KnownEnum1["JobDescriptions"] = "job_descriptions";
})(KnownEnum1 || (KnownEnum1 = {}));
/** Known values of {@link GetResponses200ContentApplicationJsonSchemaResultsItemDocumentType} that the service accepts. */
export var KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType;
(function (KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType) {
    KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType["Resumes"] = "resumes";
    KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType["JobDescriptions"] = "job_descriptions";
})(KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType || (KnownGetResponses200ContentApplicationJsonSchemaResultsItemDocumentType = {}));
/** Known values of {@link PostContentSchemaDocumentType} that the service accepts. */
export var KnownPostContentSchemaDocumentType;
(function (KnownPostContentSchemaDocumentType) {
    KnownPostContentSchemaDocumentType["Resumes"] = "resumes";
    KnownPostContentSchemaDocumentType["JobDescriptions"] = "job_descriptions";
})(KnownPostContentSchemaDocumentType || (KnownPostContentSchemaDocumentType = {}));
/** Known values of {@link Enum4} that the service accepts. */
export var KnownEnum4;
(function (KnownEnum4) {
    KnownEnum4["Resumes"] = "resumes";
    KnownEnum4["JobDescriptions"] = "job_descriptions";
})(KnownEnum4 || (KnownEnum4 = {}));
